### PR TITLE
Fix Mac CI Test

### PR DESCRIPTION
**Pull request by SmartManoj**
**What is the problem that this fixes or functionality that this introduces? Does it fix any open issues?**
https://github.com/OpenDevin/OpenDevin/pull/2556#issuecomment-2182396017

Need to unlink colima for installing dev version.

### CODE REVIEW

File: .github/workflows/run-unit-tests.yml [hunk 1]
```diff
@@ -46,6 +46,7 @@ jobs:
       - name: Install & Start Docker
         if: env.INSTALL_DOCKER == '1'
         run: |
+          brew unlink colima
```
Comment (tobitege) [inline .github/workflows/run-unit-tests.yml:49]: Could this work?
```
          if brew list colima &>/dev/null; then
            brew unlink colima
          fi
 ```

Comment (SmartManoj) [reply .github/workflows/run-unit-tests.yml:49]: It's added by default.
https://github.com/actions/runner-images/issues/6216#issuecomment-1274318785

Comment (neubig) [reply .github/workflows/run-unit-tests.yml:49]: I agree that this is better to check beforehand. Github actions might change their protocol down the line, so doing this check would future-proof our code.

It also might be good to add a comment:

```suggestion
    # Unlink colima to upgrade to the latest version 
    if brew list colima &>/dev/null; then
          brew unlink colima
          # unlinking colima dependency: go
          brew unlink go@1.21
    fi
```

**Review by tobitege — APPROVED** [review]
Let's try this

**Comment by SmartManoj** [comment]
Hanged?

image

**Comment by tobitege** [comment]
It's waiting for runners to pick the actions up, give it a few minutes

File: .github/workflows/run-unit-tests.yml [hunk 2]
```diff
@@ -46,7 +46,16 @@ jobs:
...
         if: env.INSTALL_DOCKER == '1'
         run: |
+    # Unlink colima to upgrade to the latest version 
+    if brew list colima &>/dev/null; then
+          brew unlink colima
+          # unlinking colima dependency: go
+          brew unlink go@1.21
+    fi
+          # unlinking colima dependency: go
+          brew unlink go@1.21
```
Comment (neubig) [inline .github/workflows/run-unit-tests.yml:56]: Sorry, indentation was off and unlink was shown twice.
```suggestion
          # Unlink colima to upgrade to the latest version 
          if brew list colima &>/dev/null; then
              brew unlink colima
              # unlinking colima dependency: go
              brew unlink go@1.21
          fi
```

File: .github/workflows/run-unit-tests.yml [hunk 3]
```diff
@@ -46,9 +46,14 @@ jobs:
       - name: Install & Start Docker
         if: env.INSTALL_DOCKER == '1'
         run: |
+    # Unlink colima to upgrade to the latest version 
+    if brew list colima &>/dev/null; then
```
Comment (tobitege) [inline .github/workflows/run-unit-tests.yml:50]: I think indentation needs to be changed, error on line 50 it says

**Comment by tobitege** [comment]
Looks good now! 👍

**Comment by tobitege** [comment]
@SmartManoj since you already had this comment in your other PR, maybe it's worth trying the uninstall way?
https://github.com/abiosoft/colima/issues/458#issuecomment-2171006045

**Comment by SmartManoj** [comment]
> @SmartManoj since you already had this comment in your other PR, maybe it's worth trying the uninstall way? [abiosoft/colima#458 (comment)](https://github.com/abiosoft/colima/issues/458#issuecomment-2171006045)

Yes.

**Review by neubig — APPROVED** [review]
Uninstall seems to have worked, I approve the change and thanks! It'll be nice to have this fixed.

**Pull request merged**